exercises/ex4.cpp

```diff
@@ -1,56 +1,80 @@
 /* Surprise me. */
 
-#include <stdio.h>
+#include <iostream>
+#include <limits>
+using namespace std;
+
+template <class Type>
+Type insert_and_control(Type data) //cin control (there isn't much that I can do to surprise you (without writing some crazy random code), so there you go)
+{
+  do
+  {
+    cin >> data;
+    if (cin.fail())
+    {
+      cerr << "Insert a number!" << endl;
+      cin.clear();
+      cin.ignore(numeric_limits<streamsize>::max(), '\n');
+    }
+    else
+      break;
+  } while (true);
+  return data;
+}
 
 int main()
 {
   int month;
 
   /* Input month number from user */
-  printf("Enter month number(1-12): ");
-  scanf("%d", &month);
-
-  switch (month)
+  do
   {
-  case 1:
-    printf("31 days");
-    break;
-  case 2:
-    printf("28/29 days");
-    break;
-  case 3:
-    printf("31 days");
-    break;
-  case 4:
-    printf("30 days");
-    break;
-  case 5:
-    printf("31 days");
-    break;
-  case 6:
-    printf("30 days");
-    break;
-  case 7:
-    printf("31 days");
-    break;
-  case 8:
-    printf("31 days");
-    break;
-  case 9:
-    printf("30 days");
-    break;
-  case 10:
-    printf("31 days");
-    break;
-  case 11:
-    printf("30 days");
-    break;
-  case 12:
-    printf("31 days");
+    cout << "Enter month number(1-12): " << endl;
+    month = insert_and_control(month);
+    switch (month)
+    {
+    case 1:
+      cout << "31 days" << endl;
+      break;
+    case 2:
+      cout << "28/29 days" << endl;
+      break;
+    case 3:
+      cout << "31 days" << endl;
+      break;
+    case 4:
+      cout << "30 days" << endl;
+      break;
+    case 5:
+      cout << "31 days" << endl;
+      break;
+    case 6:
+      cout << "30 days" << endl;
+      break;
+    case 7:
+      cout << "31 days" << endl;
+      break;
+    case 8:
+      cout << "31 days" << endl;
+      break;
+    case 9:
+      cout << "30 days" << endl;
+      break;
+    case 10:
+      cout << "31 days" << endl;
+      break;
+    case 11:
+      cout << "30 days" << endl;
+      break;
+    case 12:
+      cout << "31 days" << endl;
```

Comment:
Troppa ridondanza, raggruppa i casi, quelli che hanno gli stessi giorni li puoi raggruppare.

Reply:
Ammetto di non averci pensato, agisco subito

Reply:
Per questo esistono le code review

Reply:
potevi anche raggrupparli così https://github.com/Helias/git-course-exercises/blob/fead0f0222abb3daf36355762203b7777170fe0c/exercises/ex4.cpp

in realtà infatti intendevo proprio questo tipo di switch case